nemo/core/utils/cuda_python_utils.py

```diff
@@ -22,6 +22,10 @@
 
 
 def check_cuda_python_cuda_graphs_conditional_nodes_supported():
+    # for CPU-only environment we need to raise an exception, otherwise cuda-python library will fail
+    if not torch.cuda.is_available():
```

Comment:
When GPU is unavailable `python-cuda` library will fail. So, need to throw additional `EnvironmentError` exception

FYI: @galv